app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,94 @@
+class RequestsController < ApplicationController
+  include RequestsHelper
+
+  before_action :get_request, only: [:edit, :update, :update_status]
+
+  def index
+    @requests = Request.where('user_id in (?) or request_status_id = ?',
+                              @current_user.nested_subordinates << @current_user,
+                              RequestStatus.approved)
+  end
+
+  def new
+    @request = Request.new
+    @request_types = RequestType.all
+  end
+
+  def create
+    start_date = Date.strptime(params[:start_date], '%Y-%m-%d')
+    end_date = Date.strptime(params[:end_date], '%Y-%m-%d')
+    # Create a new request for every date
```

Comment:
Why not make requests have a start_date and end_date (or even date_time) instead of making a request for each date?

Reply:
Because if an employee creates a request for a week from 3-5pm each day, then a supervisor can approve all but one specific day. It makes things less complicated than having someone go and edit the request for a full week, and it gives more control over approval, even if it takes slightly longer to approve.